MyAwesomeProject/components/UserList.tsx

```diff
@@ -0,0 +1,118 @@
+import React from "react"
+import { Component } from 'react'
+import { StyleSheet, Text, View, SafeAreaView, FlatList, Alert, ActivityIndicator } from "react-native"
+import { getUsersPromise } from '../queries/Users'
+import { FetchResult } from "apollo-link";
+
+interface UserPresentation {
+    name: string,
+    email: string
+}
+
+interface UserListState {
+    users: UserPresentation[]
+    loadingIcon: boolean
+}
+
+export class UserList extends Component<{}, UserListState> {
+    constructor(props: {}) {
+        super(props)
+        this.state = {
+            users: [],
+            loadingIcon: true
+        }
+    }
+
+    private getUsersArray() {
+        getUsersPromise()
+        .then(result => this.setUsersList(result))
+        .catch(error => console.warn(error))
+        .finally(() => this.setState({loadingIcon: false}))
+    }
+
+    private setUsersList(result: FetchResult) {
+      if (!result.data) { return }
+      this.setState({users: result.data.Users.nodes, loadingIcon: false})
+    }
+
+  render() {
+    return (
+      <View style={styles.root}>
+        <SafeAreaView/>
+        <Text style={styles.greeting}>
+          Lista de Usuários
+        </Text>
+
+        <View style={styles.body}>
+          <FlatList
+            data={this.state.users}
+            showsVerticalScrollIndicator={false}
+            renderItem={({item}) =>
+                <View style={styles.flatview}>
+                    <Text style={styles.name}>{item.name}</Text>
+                    <Text style={styles.email}>{item.email}</Text>
+                </View>
+            }
+            keyExtractor={item => item.email}
+          />
+        </View>
+
+        <View style={styles.loading}>
+              <ActivityIndicator
+                style={{display: this.state.loadingIcon ? "flex" : "none"}}
```

Comment:
Não sei se vale a pena renderizar o componente com `display: none`, quando for o caso 🤔.
Dá uma lida sobre Conditional Rendering